src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -41,6 +41,7 @@ public override void Write(string message)
                     base.Write(message.Substring(offset));
                 }
             }
+            public override int IndentLevel { get { return 0; } set { } } // ignores indentation inside base.Write call
```

Comment:
All Trace Listeners (including DefaulTraceListener class itself) holds the logic to indent the string.

Therefore DebugProvider class should not indent inside the Write call.
As long as IndentLevel returns 0, then base.Write(message) will make sure not to indent the string.